### PR TITLE
review TODO: this currently blocks

### PR DESCRIPTION
**Pull request by alejandraferreirovidal**
- https://github.com/jenkinsci/parameterized-remote-trigger-plugin/pull/32#discussion_r172030544
`//TODO: This currently blocks`
Where? I do not see any synchronization in your code

### CODE REVIEW

**Comment by cashlalala** [comment]
Gonna sleep first

File: src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/RemoteBuildConfiguration.java [hunk 1]
```diff
@@ -720,29 +725,44 @@ public void performWaitForBuild(BuildContext context, Handle handle)
...
-        context.logger.println("  Remote build URL: " + buildData.getURL());
-        context.logger.println("  Remote build number: " + buildData.getBuildNumber());
+        RemoteBuildInfo buildInfo = handle.getBuildInfo();
+        String queueId = buildInfo.getQueueId();
+        if (queueId == null) {
+            throw new AbortException(String.format("Unexpected status: %s. The queue id was not found.", buildInfo.toString()));
+        }
+        context.logger.println("  Remote job queue number: " + buildInfo.getQueueId());
+
+        if (buildInfo.getQueueStatus() == RemoteBuildQueueStatus.QUEUED) {
```
Comment (cashlalala) [inline src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/RemoteBuildConfiguration.java]: I'll write some utility functions like  `buildInfo.isQueued()`
`
boolean isQueued(){
  return buildInfo.getQueueStatus() == RemoteBuildQueueStatus.QUEUED;
}
`
So as the rest of comparisons of `RemoteBuildQueueStatus` & `RemoteBuildStatus`

By doing this, we can 
1. Remove the duplicate comparisons like  line 736 & 740
2. Better encapsulation

File: src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/pipeline/Handle.java [hunk 2]
```diff
@@ -109,17 +103,7 @@ public Handle(@Nonnull RemoteBuildConfiguration remoteBuildConfiguration, @Nonnu
...
-
-        PrintStreamWrapper log = new PrintStreamWrapper();
-        try {
-            //TODO: This currently blocks
-            getBuildData(queueId, log.getPrintStream());
-            return false;
-        } finally {
-            lastLog = log.getContent();
-        }
+        return buildInfo.getQueueStatus() == RemoteBuildQueueStatus.QUEUED;
```
Comment (cashlalala) [inline src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/pipeline/Handle.java]: As mentioned before, if there's  a buildInfo.isQueued() can be reused.

**Comment by alejandraferreirovidal** [comment]
Thank you very much!

**Pull request merged**